src/cluster.c

```diff
@@ -2619,7 +2785,7 @@ void clusterSendPing(clusterLink *link, int type) {
     clusterMsg *hdr;
     int gossipcount = 0; /* Number of gossip sections added so far. */
     int wanted; /* Number of gossip sections we want to append if possible. */
-    int totlen; /* Total packet length. */
+    int estlen; /* Upper bound on estimated packet length */
```

Comment:
Not sure if it's worth doing this estimate vs. always allocating a worst case: it's a short lived buffer so shouldn't impact memory but might actually help reduce fragmentation.

Reply:
My argument against this is that it adds another two functions which need to stay in sync. It shouldn't really affect fragmentation too much, since the hostname won't really change sizes.